flytekit/core/future.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import annotations
+
+import base64
+import gzip
+import hashlib
+import pathlib
+import tempfile
+import typing
+from datetime import timedelta
+from typing import TYPE_CHECKING
+
+import cloudpickle
+
+if TYPE_CHECKING:
+    from flytekit.core.base_task import PythonTask
+    from flytekit.core.workflow import WorkflowBase
+
+
+class FlyteFuture:
```

Comment:
This is really cool, I am thinking maybe we can also use the `init_remote` to set some parts of the Context that allows us to not rely on `ipython_check` of when to pickle. For example, we could simply set a new variable like `interactive_mode_enabled=True` and then this will allow the pickling?